src/Raven.Server/Documents/Queries/DocumentQueryResult.cs

```diff
@@ -80,4 +80,44 @@ private static string CreateExplanation(Lucene.Net.Search.Explanation explanatio
             return explanation.ToString();
         }
     }
+
+    public class IdsQueryResult : QueryResultServerSide<string>
+    {
+        public Queue<string> Ids { get; } = new Queue<string>();
```

Comment:
Why a queue and not a list?

Reply:
`ExecuteRateLimitedOperations` using a queue